plugins/module_utils/dnac.py

```diff
@@ -117,14 +99,36 @@ def __init__(self, module):
         self.supported_states = ["merged", "deleted", "replaced", "overridden", "gathered", "rendered", "parsed"]
         self.result = {"changed": False, "diff": [], "response": [], "warnings": []}
 
-    def get_ccc_version_as_string(self):
-        return self.dnac_version_in_string
+    def compare_dnac_versions(self, version1, version2):
+        """
+        Compare two DNAC version strings.
+
+        param version1: str, the first version string to compare (e.g., "2.3.5.3")
+        param version2: str, the second version string to compare (e.g., "2.3.7.6")
+        return: int, returns 1 if version1 > version2, -1 if version1 < version2, and 0 if they are equal
+        """
+        # Split version strings into parts and convert to integers
+        v1_parts = list(map(int, version1.split('.')))
+        v2_parts = list(map(int, version2.split('.')))
+
+        # Compare each part of the version numbers
+        for v1, v2 in zip(v1_parts, v2_parts):
+            if v1 > v2:
+                return 1
+            elif v1 < v2:
+                return -1
+
+        # If versions are of unequal lengths, check remaining parts
+        if len(v1_parts) > len(v2_parts):
+            return 1 if any(part > 0 for part in v1_parts[len(v2_parts):]) else 0
+        elif len(v2_parts) > len(v1_parts):
+            return -1 if any(part > 0 for part in v2_parts[len(v1_parts):]) else 0
 
-    def get_ccc_version_as_integer(self):
```

Comment:
is not a problem.. Don't we use this in other modules? Have you checked it?

Reply:
Updated the code accordingly.